programs/mpl-core/src/utils/mod.rs

```diff
@@ -226,6 +228,32 @@ pub(crate) fn validate_asset_permissions<'a>(
         }
     };
 
+    let plugin_set: HashSet<_> =
+        if asset.data_len() > deserialized_asset.len() {
+            let plugin_list = list_plugins::<AssetV1>(asset)?;
+            plugin_list.into_iter().collect()
+        }
+        else {
+            HashSet::new()
+        };
+    
+    if plugin_set.contains(&PluginType::UpdateDelegate) {
```

Comment:
I think you can push this logically into the UpdateDelegate plugin for the Update lifecycle event. It'll only be checked if the UpdateDelegate exists and the reject will always override a success. I think it should be logically equivalent to what you're doing here and you won't have to change our messy validation function, haha.